vllm/model_executor/layers/quantization/gguf.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+from gguf.constants import GGML_QUANT_SIZES
+from torch.nn.parameter import Parameter, UninitializedParameter
+
+from vllm import _custom_ops as ops
+from vllm.model_executor.layers.linear import LinearBase, LinearMethodBase
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+from vllm.model_executor.layers.vocab_parallel_embedding import (
+    VocabParallelEmbedding)
+from vllm.model_executor.utils import set_weight_attrs
+
+
+class GGUFConfig(QuantizationConfig):
+    """Config class for GGUF."""
+
+    def __init__(self, ) -> None:
+        pass
+
+    def __repr__(self) -> str:
+        return ("GGUFConfig()")
+
+    @property
+    def merge_weight(self) -> bool:
+        return False
+
+    @property
+    def is_neox_style(self) -> bool:
+        return False
+
+    def get_name(self) -> str:
+        return "gguf"
+
+    def get_supported_act_dtypes(self) -> List[torch.dtype]:
+        return [torch.half, torch.bfloat16]
+
+    @classmethod
+    def get_min_capability(cls) -> int:
+        return 60
+
+    @classmethod
+    def get_config_filenames(cls) -> List[str]:
+        return []  # no extra configs.
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "GGUFConfig":
+        return cls()
+
+    def get_quant_method(
+            self, layer: torch.nn.Module) -> Optional["GGUFLinearMethod"]:
+        if isinstance(layer, (LinearBase, VocabParallelEmbedding)):
+            return GGUFLinearMethod(self)
```

Comment:
Please separate the embedding into a different method rather than overloading GGUFLinearMethod. This will remove the needed for `apply_embeds`. This follows the patterns used in fp8 for MoE and Attention: https://github.com/vllm-project/vllm/blob/b422d4961a3052c5b4bcfc3747a1ad55acfe7eb8/vllm/model_executor/layers/quantization/fp8.py#L73-L78